docs/CLI.md

```diff
@@ -112,6 +112,65 @@ OPTIONS:
    --comment-syntax value  Comment syntax to use for snapshot files (default: "//")
 ```
 
+## `scip test`
+
+```
+NAME:
+   scip test - Validate subsets of snapshot files
+
+USAGE:
+   scip test [command options] [arguments...]
+
+DESCRIPTION:
+   The test subcommand validates whether a provided scip index
+   file matches manually specified symbol fields within an index. This can be used in
+   conjunction with snapshot to provide a cohesive testing framework for scip indexers.
+
+OPTIONS:
+   --from value            Path to SCIP index file (default: "index.scip")
+   --comment-syntax value  Comment syntax to use for snapshot files (default: "//")
+   --help, -h              show help
+```
+
+### File Specification
+
+The validation files for `scip test` are loosely based on [sublime text's](https://www.sublimetext.com/docs/syntax.html#testing) syntax highlighting files. Symbol ranges are selected using `^`, `^^` and `// <-`. Symbol attributes can be specified as the first word: `definition`, `reference`, `diagnostic`, and all subsequent lines are interpreted based on the kind of symbol attribute you are writing a test for
```

Comment:
```suggestion
The validation files for `scip test` are loosely based on [Sublime Text's](https://www.sublimetext.com/docs/syntax.html#testing) syntax highlighting files. Symbol ranges are selected using `^`, `^^` and `// <-`. Symbol attributes can be specified as the first word: `definition`, `reference`, `diagnostic`, and all subsequent lines are interpreted based on the kind of symbol attribute you are writing a test for
```

Here, `diagnostic` is technically a field and definition/reference are roles. 'symbol attributes' sounds like standard SCIP terminology, but it actually doesn't have any meaning in SCIP. Later in this doc, the same thing is referred to as 'selection kind'. Alternate terminology suggestion, say something like:

> There are 2 types of checks supported by `scip test` - for symbol roles and for diagnostics. ...
>
> The syntax to check symbol roles is as follows: \<code snippet\>
>
> The syntax to check diagnostics is as follows: \<code snippet\>

If you want, it might make sense to add a way to check/test relationships too.